### PR TITLE
nan_to_num

### LINKED ISSUE #14040: nan_to_num

### PR DESCRIPTION
**Pull request by Dhruv2012**
Close #14040

### CODE REVIEW

**Comment by Dhruv2012** [comment]
@Fayad-Alman, relevant test cases are passing.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py [hunk 1]
```diff
@@ -2374,3 +2374,44 @@ def test_torch_sgn(
         input=input[0],
         out=None,
     )
+
+
+@handle_frontend_test(
+    fn_tree="torch.nan_to_num",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py:2382]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```

Please use valid dtypes instead of strictly using floats

Comment (Dhruv2012) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py:2382]: Hi @ZiadAmerr, Thank you for the review! I saw that valid dtype includes complex dtypes which are not inherently supported for the API. 

Further, I took reference from the **nan_to_num** API  for other frontends (numpy, JAX), which have used **float** dtype for testing (probably due to the fact that complex types are not supported). Also, the corresponding experimental API seems to use float.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py [hunk 2]
```diff
@@ -2374,3 +2374,44 @@ def test_torch_sgn(
...
+        min_num_dims=1,
+        max_num_dims=3,
+        min_value=-100,
+        max_value=100,
+        allow_nan=True,
+        allow_inf=True,
+    ),
+    nan=st.floats(min_value=0.0, max_value=100),
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py:2390]:
```suggestion
    nan=st.floats(min_value=-100.0, max_value=100.0),
```

Let's also allow for negative values here

File: ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py [hunk 3]
```diff
@@ -2374,3 +2374,44 @@ def test_torch_sgn(
...
+    posinf=st.floats(min_value=5e100, max_value=5e100),
+    neginf=st.floats(min_value=-5e100, max_value=-5e100),
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py:2392]: Please add possibility to test if the function behaves correctly if `None` is passed to either `posinf` or `neginf`

File: ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py (hunk 1 above)
Comment (ZiadAmerr) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py:2382]: I will look into that and get back to you as soon as possible :)
But feel free to take a look at `get_dtypes(...)` documentation as follows:

> Draws a valid dtypes for the test function. For frontend tests, it draws the datatypes from the intersection between backend framework data types and frontend framework dtypes, otherwise, draws it from backend framework data types.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py (hunk 2 above)
Comment (Dhruv2012) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py]: hi @ZiadAmerr, modified the test cases in latest PR commit here to accept negative values.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py (hunk 3 above)
Comment (Dhruv2012) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_pointwise_ops.py]: Also, modified the posinf and neginf to accept None in the tests

**Comment by Dhruv2012** [comment]
Hii @ZiadAmerr,
Thanks for the detailed review once again. I modified the tests to accept **negative values** for **NaN** and **None** for **posinf** and **neginf**. The relevant tests seem to be passing (run_tests_1 is not passing due to a previous dev of test_jax_numpy_hamming). Hopefully, it can be merged now!

**Comment by ZiadAmerr** [comment]
LGTM, thanks @Dhruv2012 :)
